pcdsdevices/lens.py

```diff
@@ -286,31 +341,294 @@ def _make_safe(self):
+
+    For this documentation purposes we will use the SimLensStack:
+
+    >>> sim = lens.SimLensStack(path=path, x_prefix='x', y_prefix='y',
+                                z_prefix='z', z_offset=2, z_dir=0.2, E=8,
+                                att_obj=att, name='sim')
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+
+    You can check what current sets are in the file as follows:
+
+    >>> sim.read_lens(print_only=True)
+    [[3, 0.0001, 1, 0.0002], [1, 0.0001, 1, 0.0003, 1, 0.0005],
+    [2, 0.0001, 1, 0.0005]]
+
+    If `print_only` is `False`, `self.lens_pack` will be set to use the
+    current sets:
+
+    >>> sim.read_lens()
+    [[3, 0.0001, 1, 0.0002],
+    [1, 0.0001, 1, 0.0003, 1, 0.0005],
+    [2, 0.0001, 1, 0.0005]]
+
+    >>> sim.lens_pack
+    [[3, 0.0001, 1, 0.0002],
+    [1, 0.0001, 1, 0.0003, 1, 0.0005],
+    [2, 0.0001, 1, 0.0005]]
+
+    You can also create the lens sets using this class like so:
+
+    >>> sets_list = [[3, 0.0001, 1, 0.0002],
+                     [1, 0.0001, 1, 0.0003, 1, 0.0005],
+                     [2, 0.0001, 1, 0.0005]]
+    >>> sim.create_lens(sets_list)
+
+    For calculations, one set at the time will be used. Use the `set_lens_set`
+    method to choose what set to use. Ex. to use the second set:
+
+    >>> sim.set_lens_set(2)
+
+    >>> sim.lens_set
+    [1, 0.0001, 1, 0.0003, 1, 0.0005]
+
+    Now you can try to tweak the motors:
+
+    >>> sim.tweak()
+    sim_x: -0.1000, sim_y: 0.2000, scale: 0.1
+    Left: move x motor left
+    Right: move x motor right
+    Down: move y motor down
+    Up: move y motor up
+    + or Shift_Up: scale*2
+    - or Shift_Down: scale/2
+    Press q to quit. Press any other key to display this message.
+
+    You can then call the `align()` function to create presets
+
+    >>> be_stack.align()
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    sim_x: -0.1000, sim_y: 0.2000, scale: 0.1
+
+    Check the positions of the motors like so:
+
+    >>> sim.x
+    sim_x
+    -----
+    preset: align_position_one
+    position: -0.1
+
+    >>> sim.calib_z()
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    80
+
+    >>> sim.beam_size()
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+    0.00024626624937199417
+
+    Use `calib_z` and `beam_size` to to move the z motor:
```

Comment:
Is this correct? we use `calib_z` and `beam_size` to move the z motor?

Reply:
Right now, if you move `calib_z` it will move the x, y, and z motors at the same time based on the last time we ran the `align` method. `beam_size` will do a calculation on top of that and move the `calib_z` motor underneath, so you can do `beam_size.move(size)` and it will move x, y, and z appropriately to get to the correct spot size at the sample.